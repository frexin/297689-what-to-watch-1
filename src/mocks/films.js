const movies = [
  {
    id: 1, name: `Bohemian Rhapsody`, picture: `bohemian-rhapsody.jpg`, genre: `Crime`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2, name: `Aviator`, picture: `aviator.jpg`, genre: `Dramas`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`

  },
  {
    id: 3, name: `What We Do in the Shadows`, picture: `what-we-do-in-the-shadows.jpg`, genre: `Crime`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4, name: `Snatch`, picture: `snatch.jpg`, genre: `Horror`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 5, name: `Revenant`, picture: `revenant.jpg`, genre: `Romance`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

export {movies};
