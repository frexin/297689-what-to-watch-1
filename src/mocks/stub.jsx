import React from "react";

const StubComponent = () => {
  return (<div/>);
};

export default StubComponent;
