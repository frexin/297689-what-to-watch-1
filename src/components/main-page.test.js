import React from 'react';
import renderer from 'react-test-renderer';
import MainPage from './main-page.jsx';

import {movies} from "../mocks/films";

it(`Main components loads correctly`, () => {
  const tree = renderer.create(<MainPage movies={movies} />).toJSON();

  expect(tree).toMatchSnapshot();
});
