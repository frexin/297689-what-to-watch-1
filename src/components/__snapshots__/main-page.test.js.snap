// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main components loads correctly 1`] = `
<div>
  <div
    className="page-content catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Bohemian Rhapsody"
          height={175}
          src="img/bohemian-rhapsody.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Aviator"
          height={175}
          src="img/aviator.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
        >
          Aviator
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="What We Do in the Shadows"
          height={175}
          src="img/what-we-do-in-the-shadows.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
        >
          What We Do in the Shadows
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Snatch"
          height={175}
          src="img/snatch.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
        >
          Snatch
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Revenant"
          height={175}
          src="img/revenant.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
        >
          Revenant
        </a>
      </h3>
    </article>
  </div>
</div>
`;
